backend/infrahub/core/task/task.py

```diff
@@ -91,3 +108,83 @@ async def query(
             )
 
         return {"count": count, "edges": nodes}
+
+
+class NewTask:
+    @classmethod
+    async def query(
+        cls,
+        fields: dict[str, Any],
+        related_nodes: list[str],
+        branch: str | None = None,
+        limit: int | None = None,
+        offset: int = 0,
+    ) -> dict[str, Any]:
+        nodes: list[dict] = []
+        count = None
+
+        log_fields = get_nested_dict(nested_dict=fields, keys=["edges", "node", "logs", "edges", "node"])
+        logs_flow: dict[str, list[PrefectLog]] = defaultdict(list)
+
+        async with get_client(sync_client=False) as client:
+            tags = [TAG_NAMESPACE]
+
+            if branch:
+                tags.append(WorkflowTag.BRANCH.render(identifier=branch))
+
+            # We only support one related node for now, need to investigate HOW (and IF) we can support more
+            if related_nodes:
+                tags.append(WorkflowTag.RELATED_NODE.render(identifier=related_nodes[0]))
+
+            flow_run_filters = FlowRunFilter(
+                tags=FlowRunFilterTags(all_=tags),
+            )
+
+            flows = await client.read_flow_runs(
+                flow_run_filter=flow_run_filters,
+                limit=limit,
+                offset=offset,
+                sort=FlowRunSort.START_TIME_DESC,
+            )
+
+            # For now count will just return the number of objects in the response
+            # it won't work well with pagination but it doesn't look like Prefect provide a good option to count all flows
+            if "count" in fields:
+                count = len(flows)
+
+            if log_fields:
+                flow_ids = [flow.id for flow in flows]
+                all_logs = await client.read_logs(log_filter=LogFilter(flow_run_id=LogFilterFlowRunId(any_=flow_ids)))
+                for log in all_logs:
+                    logs_flow[log.flow_run_id].append(log)
+
+            for flow in flows:
+                logs = []
+                if log_fields:
+                    logs = [
+                        {
+                            "node": {
+                                "message": log.message,
+                                "severity": LOG_LEVEL_MAPPING.get(log.level, "error"),
+                                "timestamp": log.timestamp,
+                            }
+                        }
+                        for log in logs_flow[flow.id]
+                    ]
+
+                nodes.append(
+                    {
+                        "node": {
+                            "title": flow.name,
+                            "conclusion": flow.state_name,
+                            "related_node": "",
+                            "related_node_kind": "",
+                            "created_at": flow.created.to_day_datetime_string(),
+                            "updated_at": flow.updated.to_day_datetime_string(),
```

Comment:
Just a question about these would it be the same type of time format that we're using internally? If not I think we should change or create an issue to ensure that all timestamps coming out from Infrahub look the same.

Reply:
Good catch, I fixed that, we're using iso8601 everywhere